lib/ansible/modules/cloud/misc/virt.py

```diff
@@ -450,6 +461,11 @@ def core(module):
             module.fail_json(msg = "state change requires a guest specified")
 
         res['changed'] = False
```

Comment:
We could remove the default value of `autostart` option, then:

- when `autostart` option isn't used, the value of `autostart` parameter will be `None`
- when `autostart` option used, `Virt.autostart` need to be called only when the current configuration differs from the wanted configuration